pkg/proxy/proxy.go

```diff
@@ -284,10 +297,19 @@ func (p *Proxy) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		case <-errc:
 			// Only wait for a single error and let the defers close both connections.
 			return
-		case <-ticker.C:
+		case <-timeoutTimer.C:
+			// Close websocket after a reasonable timeout to prevent connection leaks
```

Comment:
This might be good to do anyway since long-lived websockets can keep the haproxy pods from terminating, but I'm not sure if this is the cause of the connection leak. I'd think this would be a problem even in single cluster otherwise.